Gridy/Controller/PuzzleViewController.swift

```diff
@@ -35,22 +46,90 @@ class PuzzleViewController: UIViewController, UIGestureRecognizerDelegate {
         self.newGameButton.clipsToBounds = true
     }
     
-    @IBAction func startNewGame(_ sender: Any) {
-        dismiss(animated: true)
-    }
-    
     private func setupPuzzlePiecesImageViews() {
         let randomizedImagePieces = imagePieces.shuffled()
         
+        //I want to apply to the puzzle pieces the same frames as the pieces placeholders
+        for puzzlePiece in puzzlePiecesImageViews {
+            for puzzlePiecePlaceholder in puzzlePiecesPlaceholdersViews {
+                puzzlePiece.frame = puzzlePiecePlaceholder.frame
+            }
+        }
```

Comment:
You can use https://developer.apple.com/documentation/swift/1541125-zip
```
zip(puzzlePiecesImageViews, puzzlePiecesPlaceholdersViews).forEach {
    // your code
}
```
Or you can use array indexes, and you are already using them below:
```
puzzlePiecesImageViews[index]
puzzlePiecesPlaceholdersViews[index]
```